Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -7204,15 +7215,17 @@ class BBQJIT {
             m_jit.vectorTruncSat(info, valueLocation.asFPR(), resultLocation.asFPR());
 #endif
             return { };
-        case JSC::SIMDLaneOperation::Not:
+        case JSC::SIMDLaneOperation::Not: {
 #if CPU(X86_64)
-                // Equality is always a single instruction for this lane, so it doesn't matter that our scratch register is the same as the operands.
-                m_jit.compareIntegerVector(RelationalCondition::Equal, SIMDInfo { SIMDLane::i32x4, SIMDSignMode::None }, m_scratchFPR, m_scratchFPR, m_scratchFPR, m_scratchFPR);
-                m_jit.vectorXor(info, valueLocation.asFPR(), m_scratchFPR, resultLocation.asFPR());
+            // Equality is always a single instruction for this lane, so it doesn't matter that our scratch register is the same as the operands.
+            ScratchScope<0, 1> scratches(*this, valueLocation, resultLocation);
+            m_jit.compareIntegerVector(RelationalCondition::Equal, SIMDInfo { SIMDLane::i32x4, SIMDSignMode::None }, m_scratchFPR, m_scratchFPR, m_scratchFPR, scratches.fpr(0));
```

Comment:
Has this actually caused any bugs? The scratch parameter to `compareIntegerVector` can be clobbered generally, but for `RelationalCondition::Equal` specifically, we shouldn't use it (hence the comment I left above). So I figured we might as well not avoid an extra unnecessary register.

Reply:
I think it is unsafe that passing the same register as a scratchFPR to MacroAssembler. MacroAssembler is thinking that scratchFPR is separate from the other registers and do not assume that the above is safe.

Reply:
Let's remove that comment since it is incorrect.

Reply:
Removed

Reply:
I guess my thinking is, I can't imagine us needing to change the macro assembler implementation, so it's probably not too problematic to assume this behavior (since we know the architecture and instruction we are generating exactly). Whereas reserving an extra register could cause us to spill unnecessarily. But I suppose this change is fine.